# HG changeset patch
# User Geoff Lankow <geoff@darktrojan.net>
# Date 1696443573 -7200
# Node ID bea7459df29b506ebeb5607d41cc5d43e78e6c7f
# Parent  21921c58cb440ecc14a6a2961b0c5b19561bf166
Bug 1854422 - Set ICAL.design.strict to false in the main thread. r=mkmelin

Setting this property disappeared when libical was removed. I've added it to calUtils.jsm because that should get loaded before anything tries to use ical.js.

Differential Revision: https://phabricator.services.mozilla.com/D189603

diff --git a/calendar/base/modules/utils/calItipUtils.jsm b/calendar/base/modules/utils/calItipUtils.jsm
--- a/calendar/base/modules/utils/calItipUtils.jsm
+++ b/calendar/base/modules/utils/calItipUtils.jsm
@@ -12,16 +12,19 @@
 const EXPORTED_SYMBOLS = ["calitip"];
 
 var { MailServices } = ChromeUtils.import("resource:///modules/MailServices.jsm");
 var { calendarDeactivator } = ChromeUtils.import(
   "resource:///modules/calendar/calCalendarDeactivator.jsm"
 );
 var { XPCOMUtils } = ChromeUtils.importESModule("resource://gre/modules/XPCOMUtils.sys.mjs");
 
+const { ICAL } = ChromeUtils.import("resource:///modules/calendar/Ical.jsm");
+ICAL.design.strict = false;
+
 const lazy = {};
 XPCOMUtils.defineLazyModuleGetters(lazy, {
   CalAttendee: "resource:///modules/CalAttendee.jsm",
   CalRelation: "resource:///modules/CalRelation.jsm",
   CalItipDefaultEmailTransport: "resource:///modules/CalItipEmailTransport.jsm",
   CalItipMessageSender: "resource:///modules/CalItipMessageSender.jsm",
   CalItipOutgoingMessage: "resource:///modules/CalItipOutgoingMessage.jsm",
 });
